YoutubeExplode/Internal/YoutubeVideoStream.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace YoutubeExplode.Internal
+{
+    internal class YoutubeVideoStream : Stream
```

Comment:
Since it's not directly tied to Youtube in any way, can we call it `SegmentedHttpStream` or something?
It would also be nice if there was an extension method that would let me do `httpClient.GetSegmentedAsync(url)`.

Reply:
Renamed, and added an extensions.